packages/typespec-client-generator-core/src/lib.ts

```diff
@@ -315,13 +315,7 @@ export const $lib = createTypeSpecLibrary({
     "invalid-alternate-source-type": {
       severity: "error",
       messages: {
-        default: paramMessage`@alternateType only supports scalar types. The source type is '${"typeName"}'.`,
-      },
-    },
-    "invalid-alternate-type": {
-      severity: "error",
-      messages: {
-        default: paramMessage`@alternateType only supports scalar types, scalar array or unknown as the alternate type.`,
+        default: paramMessage`Can't apply none-Scalar type alternate to Scalar type source. The alternate type kind is '${"kindName"}'`,
```

Comment:
can you help to rephase the word? it is not easy to understand.

Reply:
the diagnostic message has been rephased.